modules/unsupported/geojson-core/src/main/java/org/geotools/data/geojson/GeoJSONReader.java

```diff
@@ -411,8 +427,7 @@ private SimpleFeature getNextFeature(ObjectNode node) throws IOException {
                 }
             }
             builder.set(GEOMETRY_NAME, g);
-
-            String newId = baseName + "." + nextID++;
+            String newId = getId(node);
```

Comment:
If we are using the `ID` attribute then it should be excluded from the feature that is created.

Reply:
Yup, ambiguous method name here.  Changed to reflect that it either gets the ID or generates one if not found.

Reply:
I was saying that if we use the `id` attribute as the feature ID then we should not return it as an attribute too

Reply:
Yup, my mistake.  Fixed to exclude id field from general attribute map.